### PR TITLE
GATEWAY 2948

### PR DESCRIPTION
**Pull request by zmelnick**
Create classes that extend BaseEventDescriptionBuilder for each service.

Unit test.

### CODE REVIEW

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import gov.hhs.fha.nhinc.event.EventContextAccessor;
+import gov.hhs.fha.nhinc.event.MessageRoutingAccessor;
+
+/**
+ * @author zmelnick
+ *
+ */
+public class EDXLDistributionDescriptionBuilder extends BaseEventDescriptionBuilder {
+
+    private EDXLDistribution alert;
```
Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: Recommend this be final, and all constructors require it to be passed in.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+     */
+    @Override
+    public void buildPayloadSize() {
+        if (alert != null) {
+            List<ContentObjectType> contents = alert.getContentObject();
+            BigInteger payloadSize = null;
+            for (ContentObjectType message : contents) {
+                if (message.getXmlContent() != null
+                        || (message.getNonXMLContent() != null && message.getNonXMLContent().getSize() == null)) {
+                    LOG.warn("Paylod size not provided");
```
Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: Warn seems like an excessively high level.  Recommend info, or lower.  This is happening inside a loop.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+     *
+     * @see gov.hhs.fha.nhinc.event.EventDescriptionBuilder#buildPayloadSize()
+     */
+    @Override
+    public void buildPayloadSize() {
+        if (alert != null) {
+            List<ContentObjectType> contents = alert.getContentObject();
+            BigInteger payloadSize = null;
+            for (ContentObjectType message : contents) {
+                if (message.getXmlContent() != null
```
Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: Complicated if predicate, perform extract method.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                        || (message.getNonXMLContent() != null && message.getNonXMLContent().getSize() == null)) {
+                    LOG.warn("Paylod size not provided");
+                    payloadSize = null;
+                    break;
+                } else if (payloadSize != null) {
```
Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: You can remove one of the branches here by initializing payloadSize to 0 at the start.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                    payloadSize = payloadSize.add(message.getNonXMLContent().getSize());
+                } else {
+                    payloadSize = message.getNonXMLContent().getSize();
+                }
+            }
+            if (payloadSize != null) {
```
Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: And then change the check here to be != 0.

File: Product/Production/Services/AdminDistributionCore/src/test/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilderTest.java [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+    /**
+     *
+     */
+    private static final String INITIATING_HCID = "1.1";
+
+    /**
+     *
+     */
+    public EDXLDistributionDescriptionBuilderTest() {
```
Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/test/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilderTest.java]: Remove - this isn't doing anything.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see gov.hhs.fha.nhinc.event.EventDescriptionBuilder#buildTimeStamp()
+     */
+    @Override
+    public void buildTimeStamp() {
+    }
```
Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java:64]: probably worth a comment here that we're still sorting out the timestamp extraction.

**Comment by tparnell** [comment]
Would like to see the issues I identified fixed prior to merge.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java (hunk 4 above)
Comment (zmelnick) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: BigInteger is immutable, and the only way to initialize it to a specific value is by passing a String decimal value in.

Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: BigInteger payloadSize = BigInteger.ZERO;
for (theContentObject) {
  if (stuffIsNull) { break }
  payloadSize = payloadSize.add(message.getNonXMLContent().getSize();
}

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java (hunk 5 above)
Comment (zmelnick) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: != 0 would not work on the BigInteger object type.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java (hunk 7 above)
Comment (zmelnick) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java:64]: will do.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java (hunk 5 above)
Comment (tparnell) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/aspect/EDXLDistributionDescriptionBuilder.java]: Yes.  That's a logical !=.  The actual code is (I believe):
if (BigInteger.ZERO.compareTo(payloadSize) == 0) ...

**Pull request merged**